amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseRefresher.java

```diff
@@ -292,7 +292,13 @@ long sleep(long timeToSleepMillis) {
     @Override
     public boolean isLeaseTableEmpty()
             throws DependencyException, InvalidStateException, ProvisionedThroughputException {
-        return list(1, 1, null).isEmpty();
+        return isLeaseTableEmptyForStreamIdentifier(null);
+    }
+
+    @Override
+    public boolean isLeaseTableEmptyForStreamIdentifier(StreamIdentifier streamIdentifier)
+            throws DependencyException, ProvisionedThroughputException, InvalidStateException {
+        return list(1, 1, streamIdentifier).isEmpty();
```

Comment:
Can you confirm that when we use max page of 1, does ddb guarantee the entire table be scanned when no entries for this stream found?